### ISSUE TITLE
Does this generate TPAR reports in Australian standard?

### ISSUE THREAD

**Issue by Joe23232** [darcys22/tpar#4]
Hi, I wanted to know if this generates TPAR reports which follows Australian standard?

https://softwaredevelopers.ato.gov.au/TPARspecification

https://www.ato.gov.au/Business/Reports-and-returns/Taxable-payments-annual-report/

**Comment by darcys22** [comment]
Hi Joe,

Yeah thats the one, you can try it out here: https://www.taxablepayments.com.au/

**Comment by Joe23232** [comment]
@darcys22 Hey man, do you own that website? Did you create the website? Is it ok I can see in the source code where and how it generates the TPAR report please?

**Comment by darcys22** [comment]
Hi Joe, this repo is the source code for that website. Confirming that i created that.

feel free to peruse the source code all you wish

**Comment by Joe23232** [comment]
@darcys22 Hey man,

This sounds pretty good mate :) I just have one last question, does the website generate an electronic form which follows [this]( https://softwaredevelopers.ato.gov.au/TPARspecification) format which can be submitted to the ATO?

**Comment by darcys22** [comment]
Yep the tpar file which can be uploaded via the ATO's file transfer

**Comment by Joe23232** [comment]
@darcys22 I see that is pretty cool, unlike most other providers I am glad you are free and open source :)

Just out of curiosity, how does it generate the TPAR electronic form? How does it all work? I have been trying to read its manual page but I am still a little confused by it. I was wondering if I could please get some clarification on it?

**Comment by darcys22** [comment]
Hi Joe, 

The TPAR standard you linked is simply a text file that gets uploaded via the file upload. You can open a tpar file with notepad and see its structure. All the details are in the TPAR specification but essentially each line is a contractor with some boilerplate lines at the beginning and ending of the file.

That TPAR specification is very readable and i suggest investigating it further

**Comment by Joe23232** [comment]
Thanks mate

**Comment by Joe23232** [comment]
@darcys22 hey mate, is this the specification you read:

https://softwaredevelopers.ato.gov.au/TPARspecification
